Show currency requirement details in a popover

Mayisha opened the pull request:
Fixes #2970 

## Changes proposed in this Pull Request:
Implemented the new design for the popover linked to the currency requirement pill. (design component Kofpo8jj4zDWXDsVdlSz0f-fi-6795%3A10109)

**Before**

![before pop](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/35a01dfe-d4b0-47f8-8339-38e7a3c1cbc8)

**After**

![after pop](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/1e61993f-6a44-4342-8c1a-96ed60ab454c)


## Testing instructions
- Set store currency to USD.
- Go to the Stripe settings page and check the payment methods.
- Confirm that some of the payment methods like giropay, EPS, OXXO, Boleto etc. have a pill beside their name `Requires currency`.
- Click on the `info` icon in the currency requirement pill and confirm that a popover is displayed. The popover should have the new design. 
- Click outside and the popover should be hidden.

![popup](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/ff285b05-b714-4b4b-b0a7-fa4d6de086f5)

Mayisha committed:
add changelog

Mayisha committed:
create popover component

Mayisha committed:
use popover to display details about currency requirement

Mayisha committed:
fix popover width in mobile view

Mayisha committed:
Merge branch 'develop' into task/currency-requirement-popover

Mayisha committed:
Merge branch 'develop' into task/currency-requirement-popover

a-danae reviewed (approved):
Things look good to me 👍  Thanks for your work here @Mayisha ! 

✅ The requirement component is only displayed when the payment method doesn't support the store currency.
✅ Clicking on the icon displays the popover with the currencies supported by the payment method.
✅ Clicking outside the popover hides it.

It wasn't intuitive for me that I should click on the icon to see the currency requirement. Wondering if having the cursor as a pointer would help, or if the popover was displayed on hover instead of on click. Either way, glad to handle it in a separate PR or to discuss it with the design team first. Happy to see this merged!

Mayisha committed:
Merge branch 'develop' into task/currency-requirement-popover

Mayisha committed:
show pointer on the icon

Mayisha commented:
Thanks @a-danae :chocolate_bar: 

> It wasn't intuitive for me that I should click on the icon to see the currency requirement. Wondering if having the cursor as a pointer would help, or if the popover was displayed on hover instead of on click.

I have added the pointer in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3135/commits/50089efd843f19e1f1d88727c147669bf52f8775.

About the popover being displayed on click, I got the following from the design and implemented it this way.

> The popover appears upon clicking the (?) icon and dismisses upon clicking anywhere outside.

a-danae commented:
Gotcha, sounds like the onclick functionality would be something to discuss with design. Thanks for the pointer update! Things look good to me.

Mayisha committed:
Merge branch 'develop' into task/currency-requirement-popover
